src/main/gui/main-window.ts

```diff
@@ -190,38 +182,78 @@ const registerEventHandlerPostSetup = (
     ipcMain.on('update-has-unsaved-changes', (_, value) => {
         hasUnsavedChanges = value;
     });
+
     let forceExit = false;
     ipcMain.on('exit-after-save', () => {
         forceExit = true;
         mainWindow.close();
     });
 
+    ipcMain.on('reboot-after-save', () => {
+        app.relaunch();
+        app.exit();
+    });
+
+    const handleUnsavedChanges = (
+        event: Electron.Event,
+        callback: () => void,
+        restartCallback?: () => void
```

Comment:
I feel like this should be more generic, like `onSave` and `onDontSave`

Reply:
done